ci-tools/host-runner/README.md

```diff
@@ -0,0 +1,255 @@
+
+```bash
+$ nix run nixpkgs#nixos-generators -- -f sd-aarch64 --flake path:.#caliptra-hostrunner0 --system aarch64-linux -o ./hostrunner-image
+...
+$ ls hostrunner-image/sd-image # A new RPI image is created in the hostrunner-image folder. De-compress with zst and then flash the SD card for the Raspberry PI.
+nixos-sd-image-24.11.20250421.9684b53-aarch64-linux.img.zst
+```
+
+Once the image is flashed the Raspberry PI should be ready to boot, assuming secrets were available when generating the SD image.
+
+## Remotely modify hostrunner
+
+You can cross-compile a NixOS system and copy binaries using SSH. Below is an example command to cross-compile the caliptra-hostrunner0.
+
+The `--target-host` takes an ssh target as a parameter.
+
+```bash
+nix run nixpkgs#nixos-rebuild -- --target-host caliptra-hostrunner0 --use-remote-sudo switch --flake path:.#caliptra-hostrunner0 --impure # "caliptra-hostrunner0" is an SSH Host in this example.
+```
+
+# Udev
+
+Udev rules are used to access USB and SD cards without root permissions. You can see this in `configuration.nix`. 
+
+If you get permission errors, you most likely need to add the device to the udev rules.
+
+Use the udev tools to figure out the `idVendor` and `idProduct` attributes of the device.
+
+```nix
+  services.udev.extraRules = ''
+    SUBSYSTEMS=="usb", ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6011", OWNER="${user}", GROUP="users"
+    SUBSYSTEMS=="usb", ATTRS{idVendor}=="04e8", ATTRS{idProduct}=="6001", OWNER="${user}", GROUP="users"
+    SUBSYSTEMS=="usb", ATTRS{idVendor}=="0424", ATTRS{idProduct}=="2640", OWNER="${user}", GROUP="users"
+    SUBSYSTEMS=="usb", ATTRS{idVendor}=="0424", ATTRS{idProduct}=="4050", OWNER="${user}", GROUP="users"
+    '';
+```
+
+# Download FPGA Image
+
+Download the latest FPGA image from [GitHub](https://github.com/chipsalliance/caliptra-sw/actions/workflows/fpga-image.yml). This job is scheduled to run once a week.
+
+This image should be copied to the host runner and stored at `$HOME/zcu104.img`.
+
+# Connect an FPGA to the host runner
+
+For an example, see `hostrunners/kir-0.nix` for 3 ZCU-104s managed with systemd. You should use Nix to manage these services so they are re-producible.
+
+## Generic instructions
+
+### RPI Setup
+
+1. Install Rust
+1. Install fpga-boss.
+    - `cargo install --git https://github.com/chipsalliance/caliptra-sw.git caliptra-fpga-boss --branch main`
+1. Install rtool
+    - `(cd ci-tools/github-runner/cmd/rtool && go build .)`
+1. Download the latest FPGA image from [GitHub](https://github.com/chipsalliance/caliptra-sw/actions/workflows/fpga-image.yml). This job is scheduled to run once a week.
+    - In the following examples, the image is stored in `$HOME/zcu-scripts/zcu104.img` for the Raspberry PI user.
+1. Get a GitHub Application private key for the CI. This is used to sign JWTs used by the FPGA runners.
```

Comment:
Can you link to GH instructions for how to do this?

Reply:
Added!